reporter/src/main/kotlin/reporters/SpdxDocumentReporter.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.reporter.reporters
+
+import org.ossreviewtoolkit.reporter.Reporter
+import org.ossreviewtoolkit.reporter.ReporterInput
+import org.ossreviewtoolkit.reporter.utils.SpdxDocumentModelMapper
+import org.ossreviewtoolkit.spdx.SpdxModelSerializer
+
+import java.io.File
+
+/**
+ * Creates YAML and JSON SPDX documents.
+ */
+class SpdxDocumentReporter : Reporter {
+    companion object {
+        const val CREATION_INFO_COMMENT = "document.comment"
+        const val DOCUMENT_COMMENT = "document.comment"
+        const val DOCUMENT_NAME = "document.name"
+        const val OUTPUT_FILE_FORMATS = "output.file.formats"
+
+        private const val DOCUMENT_NAME_DEFAULT_VALUE = "Unnamed document"
+    }
+
+    enum class FileFormat(val reportFilename: String) {
+        JSON("spdx-document.spdx.json"),
+        YAML("spdx-document.spdx.yml");
+    }
+
+    override val reporterName = "SpdxDocument"
+
+    override fun generateReport(
+        input: ReporterInput,
+        outputDir: File,
+        options: Map<String, String>
+    ): List<File> {
+        val outputFileFormats = options[OUTPUT_FILE_FORMATS]?.let {
+            it.split(",").map { token -> FileFormat.valueOf(token.toUpperCase()) }.toSet()
```

Comment:
You could use `distinct()` here to avoid the `toSet()` in the next line.

Reply:
Done.

Reply:
It works if you rewrite the code as

        val outputFileFormats = options[OUTPUT_FILE_FORMATS]
            ?.split(",")
            ?.map { token -> FileFormat.valueOf(token.toUpperCase()) }
            ?.distinct()
            ?: enumValues<FileFormat>().asList()

(This also addresses the hint of the redundant `.let` call.)

Reply:
Done.